### ISSUE TITLE
QDLDLFactorisation refactor and _factor should return Result

### ISSUE THREAD

**Issue by Lishen1** [oxfordcontrol/Clarabel.rs#49]
In the current implementation no proper way to handle ill-formed matrices. _factor_inner returns the Result but it is not passed to the caller side. just panic

**Comment by goulart-paul** [comment]
I don't really understand.   Can you perhaps make a minimal example that illustrates the failure?

**Comment by Lishen1** [comment]
for example:
```
let mut factors = QDLDLFactorisation::new(&A, Some(opts));
factors.refactor();
factors.solve(&mut b);
```
if A is not positive definite or has a deficient rank or some problem with floating point precision LDLT decomposition is not possible, so the caller side should know about failure (not panic). something like that:

```
let mut factors = QDLDLFactorisation::new(&A, Some(opts));
let result = factors.refactor();
if result.is_ok(){
   factors.solve(&mut b);
}
```

**Comment by goulart-paul** [comment]
Fixed in #53.  Will appear in v0.6 shortly.
